app/stores/ada/YoroiTransferStore.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+import { observable, action, runInAction } from 'mobx';
+import { restoreWalletForTransfer } from '../../api/ada';
+import { getReceiverAddress } from '../../api/ada/lib/storage/adaAddress';
+import { generateTransferTx } from '../../api/ada/daedalusTransfer';
+import Store from '../base/Store';
+
+import { RustModule } from '../../api/ada/lib/cardanoCrypto/rustLoader';
+
+export default class YoroiTransferStore extends Store {
+  @observable status: TransferStatus = 'gettingMnemonics';
+  @observable transferFundsRequest: Request<TransferFundsFunc>
+    = new Request<TransferFundsFunc>(this._transferFundsRequest);
+  @observable restoreForTransferRequest: Request<RestoreWalletForTransferFunc>
+    = new Request(this.api.ada.restoreWalletForTransfer);
+
+  setup(): void {
+    const actions = this.actions.ada.yoroiTransfer;
+    actions.setupTransferFundsWithMnemonic.listen(this._setupTransferFundsWithMnemonic);
+    actions.backToUninitialized.listen(this._backToUninitialized);
+    actions.transferFunds.listen(this._transferFunds);
+    actions.cancelTransferFunds.listen(this._reset);
+  }
+
+  teardown(): void {
+    super.teardown();
+    this._reset();
+  }
+
+  _setupTransferFundsWithMnemonic = async (payload: { recoveryPhrase: string }): Promise<void> => {
+    this._updateStatus('checkingAddresses');
+    // addresses and indices are one-to-one correspondences
+    const { wallet, addresses, indices } = this._restoreWalletForTransfer(recoveryPhrase);
+    const masterKey = getWalletMasterKey(wallet);
+    const cryptoWallet = getCryptoWalletFromMasterKey(masterKey, '');
+    const addressKeys = {};
+    for (var i=0; i<indices.length; i++) {
+      addressKeys[addresses[i]] = cryptoWallet.bip44_account(
+        RustModule.Wallet.AccountIndex.new(indices[i])
```

Comment:
See the `wasm` file I linked in the previous comment and the comment I made in `api/ada/index.js` on how to get all the information to regenerate the private key for each address.